src/main/java/org/openjdk/jextract/impl/UnsupportedFilter.java

```diff
@@ -161,20 +186,21 @@ public Void visitDeclaration(Declaration d, Declaration declaration) {
         return null;
     }
 
-    private boolean checkFunctionTypeSupported(Declaration decl, Type.Function func) {
+    private boolean checkFunctionTypeSupported(Declaration decl, Type.Function func, String nameOfSkipped) {
         String unsupportedType = firstUnsupportedType(func);
         if (unsupportedType != null) {
-            warn("skipping " + JavaName.getOrThrow(decl) + " because of unsupported type usage: " +
-                    unsupportedType);
+            warnUnsupportedType(nameOfSkipped, unsupportedType);
```

Comment:
Should we consolidate this with the other warnings? At the end of the day, we're skipping here because of an unsupported type, so I'd prefer to see `warnSkip` being used, where "unsupported type" appears in the "cause"

Reply:
I'll do this. It introduces a bit of duplication between the warning messages, but it's probably clearer for the reader.